src/appleseed/renderer/modeling/camera/thinlenscamera.cpp

```diff
@@ -293,11 +294,19 @@ namespace
             if (w != 1.0)
                 ray.m_org /= w;
 
-            // Compute the direction of the ray.
-            ray.m_dir.x = (point.x - 0.5) * m_kx - lens_point.x;
-            ray.m_dir.y = (0.5 - point.y) * m_ky - lens_point.y;
-            ray.m_dir.z = -m_focal_distance;
-            ray.m_dir = transform.vector_to_parent(ray.m_dir);
+            ray.m_dir = normalize(ray_direction(point.get_value(), lens_point, transform));
+
+            if (point.has_derivatives())
+            {
```

Comment:
I would rewrite this block of code (and other similar ones) as follow to minimize copy/paste errors and make correctness checking easier:

    ray.m_has_differentials = true;

    const Vector2d px(point.get_value() + point.get_dx());
    const Vector2d py(point.get_value() + point.get_dy());

    ray.m_rx.m_org = ray.m_org;
    ray.m_rx.m_dir = normalize(ray_direction(px, lens_point, transform));
    
    ray.m_ry.m_org = ray.m_org;
    ray.m_ry.m_dir = normalize(ray_direction(py, lens_point, transform));